### PR TITLE
feat: use internal WSL knobs instead of symlinks

### PR DESCRIPTION
**Pull request by K900**
This makes things so much easier and more consistent and it's undocumented API that I found in Ghidra so nothing could possibly go wrong.

### CODE REVIEW

**Comment by K900** [comment]
Filed an issue with upstream to maybe stabilize these: https://github.com/microsoft/WSL/issues/10425

**Comment by nzbr** [comment]
This looks so much better. I hope they decide to keep these

File: modules/wsl-conf.nix [hunk 1]
```diff
@@ -41,6 +41,21 @@ with lib; {
         default = false;
         description = "Use systemd as init. There's no need to enable this manually, use the wsl.nativeSystemd option instead";
       };
+      initPath = mkOption {
+        type = path;
```
Comment (SuperSandro2000) [inline modules/wsl-conf.nix:45]: Doesn't this need to be ``nullOr path``?

File: modules/wsl-distro.nix [hunk 2]
```diff
@@ -211,15 +206,20 @@ in
         (mkIf cfg.nativeSystemd {
           wsl.wslConf = {
             user.default = config.users.users.${cfg.defaultUser}.name;
-            boot.systemd = true;
+            boot = {
+              systemd = true;
+              initPath = "${nativeUtils}/bin/systemd-shim";
+              initShutdownCommand = pkgs.writeShellScript "nixos-wsl-init-shutdown" ''
+                ${pkgs.systemd}/bin/systemctl reboot
```
Comment (SuperSandro2000) [inline modules/wsl-distro.nix:213]:
```suggestion
                ${pkgs.systemd}/bin/systemctl reboot "$@"
```
just incase

File: modules/wsl-conf.nix (hunk 1 above)
Comment (K900) [reply modules/wsl-conf.nix:45]: Probably, I haven't actually tested this on non-native-systemd yet.

File: modules/wsl-distro.nix (hunk 2 above)
Comment (K900) [reply modules/wsl-distro.nix:213]: I'm actually pretty sure we can avoid the shell script here and just write the string directly.

**Comment by K900** [comment]
This is drifting out of sync and I don't want to maintain it until upstream decides this is OK to do. Closing for now, we can revisit later.

**Pull request closed without merging**